contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,818 @@
+     * @param _oracles (An array containing the address of MakerDAO's Tub contract as first element.
+     */
+    function addCollateral(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _collateral, 
+        uint256 _collateralAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        require(_collateral == ETH_TOKEN_ADDRESS, "Maker: collateral must be ETH");
+        addCollateral(_wallet, _loanId, _collateralAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Removes collateral from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _collateral The token used as a collateral (must be 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE).
+     * @param _collateralAmount The amount of collateral to remove.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     */
+    function removeCollateral(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _collateral, 
+        uint256 _collateralAmount, 
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_collateral == ETH_TOKEN_ADDRESS, "Maker: collateral must be ETH");
+        removeCollateral(_wallet, _loanId, _collateralAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Increases the debt by borrowing more token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _debtToken The token borrowed (must be the address of the DAI contract).
+     * @param _debtAmount The amount of token to borrow.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        require(_debtToken == makerCdp.sai(), "Maker: debt token must be DAI");
+        addDebt(_wallet, _loanId, _debtAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _debtToken The token to repay (must be the address of the DAI contract).
+     * @param _debtAmount The amount of token to repay.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element and Uniswap's Factory as second.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
```

Comment:
Would be nice to add `require(_oracles.length == 2)` as you did for the other providers

Reply:
same as above